libsqlite3-sys/upgrade.sh

```diff
@@ -1,28 +1,56 @@
-#!/bin/bash
+#!/bin/bash -e
 
 SCRIPT_DIR=$(cd "$(dirname "$_")" && pwd)
+CUR_DIR=$(pwd -P)
 echo "$SCRIPT_DIR"
-cd "$SCRIPT_DIR" || { echo "fatal error"; exit 1; }
-export SQLITE3_LIB_DIR=$SCRIPT_DIR/sqlite3
+cd "$SCRIPT_DIR" || { echo "fatal error" >&2; exit 1; }
+cargo clean
+mkdir -p "$SCRIPT_DIR/../target" "$SCRIPT_DIR/sqlite3" "$SCRIPT_DIR/sqlcipher"
+export SQLITE3_LIB_DIR="$SCRIPT_DIR/sqlite3"
+export SQLITE3_INCLUDE_DIR="$SQLITE3_LIB_DIR"
+export SQLCIPHER_LIB_DIR="$SCRIPT_DIR/sqlcipher"
+export SQLCIPHER_INCLUDE_DIR="$SQLCIPHER_LIB_DIR"
 
-# Download and extract amalgamation
+# Download and extract sqlite3 amalgamation
 SQLITE=sqlite-amalgamation-3340000
-curl -O https://sqlite.org/2020/$SQLITE.zip
+curl -O "https://sqlite.org/2020/$SQLITE.zip"
 unzip -p "$SQLITE.zip" "$SQLITE/sqlite3.c" > "$SQLITE3_LIB_DIR/sqlite3.c"
 unzip -p "$SQLITE.zip" "$SQLITE/sqlite3.h" > "$SQLITE3_LIB_DIR/sqlite3.h"
 unzip -p "$SQLITE.zip" "$SQLITE/sqlite3ext.h" > "$SQLITE3_LIB_DIR/sqlite3ext.h"
 rm -f "$SQLITE.zip"
 
-# Regenerate bindgen file
+# Regenerate bindgen file for sqlite3
 rm -f "$SQLITE3_LIB_DIR/bindgen_bundled_version.rs"
-export SQLITE3_INCLUDE_DIR=$SQLITE3_LIB_DIR
 cargo update
 # Just to make sure there is only one bindgen.rs file in target dir
 find "$SCRIPT_DIR/../target" -type f -name bindgen.rs -exec rm {} \;
 env LIBSQLITE3_SYS_BUNDLING=1 cargo build --features "buildtime_bindgen" --no-default-features
-find "$SCRIPT_DIR/../target" -type f -name bindgen.rs -exec cp {} "$SQLITE3_LIB_DIR/bindgen_bundled_version.rs" \;
-# Sanity check
-cd "$SCRIPT_DIR/.." || { echo "fatal error"; exit 1; }
+find "$SCRIPT_DIR/../target" -type f -name bindgen.rs -exec mv {} "$SQLITE3_LIB_DIR/bindgen_bundled_version.rs" \;
+
+SQLCIPHER_VERSION="4.4.2"
```

Comment:
It would be nice if we can upgrade only sqlite **or** sqlcipher.

Reply:
Not hard to expand the script's api --- for example, with no arguments it'd "update" both (to the versions hard-coded in the script), or with an argument of "sqlite3" it'd only "update" that one, etc. The hardest part would be settling on the interface. But this is a script that one would run 2-5 times a year, and only takes about 5 minutes to "update" both, so I assumed that extra complexity wasn't worth it.